mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -3987,6 +3987,17 @@ SmallVector<int64_t> PackOp::getStaticTiles() {
   return getStaticTilesImpl(*this);
 }
 
+SmallVector<int64_t> PackOp::getTiledOuterDims() {
+  auto innerDimsPos = getInnerDimsPos();
```

Comment:
I wonder if it should first normalize possible permutation of `inner_dims_pos` to be more intuitive.
`getAllOuterDims` is guaranteed to return dims in order w.r.t. the result shape.

Reply:
My first instinct is to get the same answers from `getAllOuterDims` and `getTiledOuterDims` for:
```
%packed = tensor.pack %arg0 outer_dims_perm = [1, 0]
    inner_dims_pos = [1, 0] inner_tiles = [8, 16]
    into %e : tensor<32x128xf32> -> tensor<16x2x8x16xf32>
```
which is not the case right now.

At the same time, I'm not certain that complicating this util with all permutation magic is worth it.
So, maybe at least some comment at the API to avoid surprises?

Reply:
Oh, I've actually missed that - thanks for pointing it out! Also, I've realised that `getAllOuterDims` should be slightly different for `PackOp` and `UnPackOp`. Sending an update shortly.